libs/designsystem/modal/src/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -621,8 +620,7 @@ export class ModalWrapperComponent
       this.routerOutlet.deactivate();
     }
     //clean up the observer
-    this.mutationObserverDisconnectFn();
-    delete this._mutationObserver;
+    this.mutationObserverUnobserveFn();
```

Comment:
Maybe this a bit more resilient by either initializing the field with a noop `(private mutationObserverUnobserveFn: UnobserveFn = () => {};)` or:
```suggestion
    this.mutationObserverUnobserveFn?.();
```

Reply:
Great idea, thanks!